resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/ServerResponseWriter.java

```diff
@@ -403,49 +418,61 @@ public int compareTo(SortableMediaType o)
          {
             return 1;
          }
+         if (this.writerType == o.writerType) return 0;
+         if (this.writerType == null) return -1;
+         if (o.writerType == null) return 1;
+         if (this.writerType.isAssignableFrom(o.writerType)) return -1;
+         if (o.writerType.isAssignableFrom(this.writerType)) return 1;
          return 0;
       }
    }
    
+   
+   private static SortableMediaType mostSpecific(MediaType m1, MediaType m2)
+   {
+      Class<?> wt1 = m1 instanceof SortableMediaType ? ((SortableMediaType)m1).writerType : null;
+      Class<?> wt2 = m2 instanceof SortableMediaType ? ((SortableMediaType)m2).writerType : null;
+      return mostSpecific(m1, wt1, m2, wt2);
+   }
+   
    /**
     * m1, m2 are compatible
     */
-   private static SortableMediaType mostSpecific(MediaType m1, MediaType m2)
+   private static SortableMediaType mostSpecific(MediaType m1, Class<?> wt1, MediaType m2, Class<?> wt2)

```

Comment:
I think I have to take my (well, ours) words back with respect to the S function in 3.7.2 being the same as the one mentioned in 3.8...or at least I'm not fully convinced anymore. The behavior that previously seemed to be explained by the S functions being the same was actually caused by a bug that's fixed in the new proposal branch, so I'm leaning towards not changing the SortableMediaType because of this.